cetmix_tower_server/tests/test_variable.py

```diff
@@ -937,3 +938,157 @@ def test_validate_value(self):
             f"{variable_default_message.DEFAULT_VALIDATION_MESSAGE}",
             "Default validation message doesn't match",
         )
+
+
+class TestVariableReferenceRename(TestTowerCommon):
+    """Ensure {{ reference }} is updated everywhere after rename."""
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.variable = cls.env["cx.tower.variable"].create(
+            {
+                "name": f"Version rename {uuid.uuid4().hex[:6]}",
+                "reference": "version_test",
+            }
+        )
+        cls.command = cls.env["cx.tower.command"].create(
+            {"name": "Show version (test)", "code": "echo {{ version_test }}"}
+        )
+        cls.file = cls.env["cx.tower.file"].create(
+            {
+                "name": "version_test.txt",
+                "code": "{{ version_test }}",
+                "server_dir": "/tmp",
+            }
+        )
+
+    def test_false_references_are_ignored(self):
+        """Ensure strings resembling variables without
+        proper Jinja braces remain unchanged."""
+        # Plain text
+        cmd_plain = self.env["cx.tower.command"].create(
+            {
+                "name": "Plain",
+                "code": "print(version_test)",
+                "variable_ids": [(6, 0, [self.variable.id])],
+            }
+        )
+        # Malformed braces
+        cmd_bad = self.env["cx.tower.command"].create(
+            {
+                "name": "BadBrackets",
+                "code": "{version_test}",
+                "variable_ids": [(6, 0, [self.variable.id])],
+            }
+        )
+        # Perform rename
+        self.variable.write({"reference": "software_version"})
+        # Reload records
+        cmd_plain.invalidate_recordset()
+        cmd_bad.invalidate_recordset()
+        # Assert no change
+        self.assertEqual(cmd_plain.code, "print(version_test)")
+        self.assertEqual(cmd_bad.code, "{version_test}")
+
+    def test_multiple_occurrences_replace_all(self):
+        """All occurrences of a variable in a single field must be updated."""
+        code = "A: {{ version_test }}, B: {{  version_test   }}, C-end"
+        cmd_multi = self.env["cx.tower.command"].create(
+            {
+                "name": "Multi",
+                "code": code,
+                # Link this record to our variable so propagation includes it
+                "variable_ids": [(6, 0, [self.variable.id])],
+            }
+        )
+
+        self.variable.write({"reference": "software_version"})
+        cmd_multi.invalidate_recordset()
+        expected = "A: {{ software_version }}, B: {{  software_version   }}, C-end"
+        self.assertEqual(cmd_multi.code, expected)
+
+    def test_template_files_unchanged(self):
+        """Files created from templates should not be modified."""
```

Comment:
Why? They should be modified  if the template they belong to is modified. Please check the corresponding code.